src/Game.tsx

```diff
@@ -1,36 +1,97 @@
 import { Link, Redirect } from 'react-router-dom';
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useCallback } from 'react';
 import { Ball } from './Ball';
 import { PlayerCard } from './PlayerCard';
 
+type TimeoutResult = ReturnType<typeof setTimeout> | null;
+
 type Props = {
-  pokemons: [string, string];
-};
-type Coords = {
-  x: number;
-  y: number;
+  pokemonOne: string;
+  pokemonTwo: string;
 };
-export function Game({ pokemons }: Props): JSX.Element {
-  const [ballCoords, setBallCoods] = useState<Coords>({ x: 50, y: 50 });
-  const [player1Coord, setPlayer1Coord] = useState(0);
-  const [player2Coord, setPlayer2Coord] = useState(100);
-
-  function changePlayerCoords(e: KeyboardEvent): void {
-    console.log(e.key);
-    // Change state here
-    setBallCoods({ x: 100, y: 100 });
-    setPlayer1Coord(50);
-    setPlayer2Coord(50);
-  }
-  useEffect(() => {
-    document.addEventListener('keydown', changePlayerCoords);
+export function Game({ pokemonOne, pokemonTwo }: Props): JSX.Element {
+  const [ballCoordY] = useState(50);
+  const [ballCoordX] = useState(50);
+  const [player1Coord, setPlayer1Coord] = useState(50);
+  const [player2Coord, setPlayer2Coord] = useState(50);
+  const [isGameStarted, setIsGameStarted] = useState(false);
+  const interval = 40;
+  const step = 5;
+
+  const generateHandlePlayCoordsCallbacks = useCallback(() => {
+    let firstPlayerInt: TimeoutResult = null;
+    let secondPlayerInt: TimeoutResult = null;
+
+    const cleanHandle = () => {
```

Comment:
Issues:

1. This function should be called `handleKeyUp` to be consistent
2. It stops intervals for both players in case of `keyup` event for any key that isn't correct
3. If the intervals keep running if the component is removed